### PR TITLE
Add a spire-agent auto kubelet verification mode

### PR DESCRIPTION
**Pull request by kfox1111**

### CODE REVIEW

**Review by sabre1041 — APPROVED** [review]
LGTM tested `auto` mode on OpenShift 4.15

File: charts/spire/charts/spire-agent/templates/daemonset.yaml [hunk 1]
```diff
@@ -91,11 +91,34 @@ spec:
...
+                  exit 0
+                fi
+                {{- end }}
                 echo Could not find certificate.
                 exit 1
               fi
               {{- end }}
+              {{- if eq .Values.workloadAttestors.k8s.verification.type "auto" }}
+              echo Mode detected as hostCert.
+              {{- end }}
```
Comment (faisal-memon) [inline charts/spire/charts/spire-agent/templates/daemonset.yaml:113]: If the mode is detected to host cert does it need to do the checks from lines 81-92?

File: charts/spire/charts/spire-agent/README.md [hunk 2]
```diff
@@ -87,7 +87,7 @@ A Helm chart to install the SPIRE agent.
...
 | `workloadAttestors.k8s.enabled`                        | Enables the Kubernetes workload attestor                                                                                                                                                          | `true`                                                                           |
-| `workloadAttestors.k8s.verification.type`              | What kind of verification to do against kubelet. Valid options are [hostCert, apiServerCA, skip]                                                                                                  | `skip`                                                                           |
+| `workloadAttestors.k8s.verification.type`              | What kind of verification to do against kubelet. Valid options are [auto, hostCert, apiServerCA, skip]                                                                                            | `skip`                                                                           |
```
Comment (faisal-memon) [inline charts/spire/charts/spire-agent/README.md:90]:
```suggestion
| `workloadAttestors.k8s.verification.type`              | What kind of verification to do against kubelet. auto will first attempt to use the apiServerCA, and then fall back on hostCert. Valid options are [auto, hostCert, apiServerCA, skip]                                                                                            | `skip`                                                                           |
```

**Comment by faisal-memon** [comment]
Tested and works on colima on mac.

File: charts/spire/charts/spire-agent/README.md (hunk 2 above)
Comment (kfox1111) [reply charts/spire/charts/spire-agent/README.md:90]: Its the other way around. Will update though with that pattern. It looks good.

Comment (kfox1111) [reply charts/spire/charts/spire-agent/README.md]: Updated. Please re-review

**Review by faisal-memon — APPROVED** [review]

**Comment by faisal-memon** [comment]
Thanks @kfox1111 for this change.

File: charts/spire/charts/spire-agent/README.md (hunk 2 above)
Comment (faisal-memon) [reply charts/spire/charts/spire-agent/README.md]: looks good.

File: charts/spire/charts/spire-agent/templates/daemonset.yaml (hunk 1 above)
Comment (faisal-memon) [reply charts/spire/charts/spire-agent/templates/daemonset.yaml:113]: nm. misunderstood code.

**Pull request merged**